src/main/java/org/folio/fqm/service/FqlToSqlConverterService.java

```diff
@@ -218,4 +245,19 @@ private static Field<Object> field(FieldCondition<?> condition, EntityType entit
       .map(SqlFieldIdentificationUtils::getSqlFilterField)
       .orElseThrow(() -> new ColumnNotFoundException(entityType.getName(), columnName));
   }
+
+  /**
+   * Build a mapping from a condition class to a condition handler.
+   * Note: This method exists in order to provide some type-safety around sqlConverters, to ensure the class and condition
+   * handler are for matching implementations of FqlCondition, since we can provide more powerful type constraints on
+   * methods than we can on an object
+   */
+  private static <T extends FqlCondition<?>> Map.Entry<Class<T>, ? extends ConditionHandler<T>> buildMapping(Class<T> conditionClass, ConditionHandler<T> f) {
+    return Map.entry(conditionClass, f);
+  }
+
+  @FunctionalInterface
+  private interface ConditionHandler<T extends FqlCondition<?>> {
+    Condition handle(T fqlCondition, EntityType entityType, Field<Object> field);
+  }
```

Comment:
Do you guys think this interface deserves a comment? It's so simple that I'm not sure if that would just be noise or not...

Reply:
I think it's good without!